src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -162,6 +162,16 @@ internal sealed class MeterProviderSdk : MeterProvider
                             try
                             {
                                 metricStreamConfig = viewConfig(instrument);
+
+                                if (metricStreamConfig is ExplicitBucketHistogramConfiguration
+                                    && instrument.GetType().GetGenericTypeDefinition() != typeof(Histogram<>))
+                                {
+                                    OpenTelemetrySdkEventSource.Log.MetricViewIgnored(
+                                        instrument.Name,
+                                        instrument.Meter.Name,
+                                        "Histogram metric stream configuration can not be used for instruments that are not histograms.",
```

Comment:
"The current SDK does not allow aggregating non-Histogram instruments as Histograms.." - Would this look nicer?

Reply:
Yes, agreed that's nicer.